pkg/oci/pusher/pusher.go

```diff
@@ -172,10 +167,18 @@ func (p *Pusher) Push(ctx context.Context, artifactType oci.ArtifactType,
 		return nil, err
 	}
 	defer rootReader.Close()
-	// Tag the root descriptor remotely.
-	err = repo.PushReference(ctx, *rootDesc, rootReader, repo.Reference.Reference)
-	if err != nil {
-		return nil, err
+
+	// Using ":latest" by default if no tag was provided.
+	if repo.Reference.Reference == "" && len(o.Tags) == 0 {
+		repo.Reference.Reference = oci.DefaultTag
+	}
+
+	if repo.Reference.Reference != "" {
+		// Tag the root descriptor remotely.
+		err = repo.PushReference(ctx, *rootDesc, rootReader, repo.Reference.Reference)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
What if you pass a ref without tag (repo.Reference.Reference == "") and len(o.Tags) != 0?
The subsequent TagN function at line 187 will fail because an empty ref is passed. We should find a way to handle this

Reply:
oops, I'll take a closer look at `TagN` and update